src/protocol.rs

```diff
@@ -108,13 +108,92 @@ pub struct Parameters<Pk, Qk, Rk, Sk, Addr, Ti, F, Pr> {
     pub spend: Qk,
     pub extra_accordant_keys: Vec<TaggedElement<u16, Qk>>,
     pub accordant_shared_keys: Vec<TaggedElement<SharedKeyId, Sk>>,
-    pub proof: Pr,
+    pub proof: Option<Pr>,
```

Comment:
This is to make it possible to pass the proof separately, right?

Reply:
Yes, commit [1e2fcb1](https://github.com/farcaster-project/farcaster-core/pull/275/commits/1e2fcb19f149b345e90b2126eedb3e042e2f0086). Because of p2p length restriction we split the reveal in two messages, the last one being the proof. So we need an option here because we create this struct on reveal and update on reveal proof in the node.

Reply:
Exactly what I was thinking, ACK.
After the next internet2 bump, this split into two messages should be reverted.